frontend/src/main/web/lib/components/glossary/SourceActionCell.jsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import {PureRenderMixin} from 'react/addons';
+import { Button, Icon, Tooltip, OverlayTrigger } from 'zanata-ui';
+import Actions from '../../actions/GlossaryActions';
+import LoadingCell from './LoadingCell'
+import DeleteEntry from './DeleteEntry'
+import GlossaryStore from '../../stores/GlossaryStore';
+import _ from 'lodash';
+
+var SourceActionCell = React.createClass({
+  propTypes: {
+    resId: React.PropTypes.string.isRequired,
+    info: React.PropTypes.string.isRequired,
+    rowIndex: React.PropTypes.number.isRequired,
+    srcLocaleId: React.PropTypes.string.isRequired,
+    canUpdateEntry: React.PropTypes.bool,
+    canDeleteEntry: React.PropTypes.bool
+  },
+
+  mixins: [PureRenderMixin],
+
+  getInitialState: function() {
+    return this._getState();
+  },
+
+  _getState: function () {
+    return {
+      entry: GlossaryStore.getEntry(this.props.resId)
+    }
+  },
+
+  componentDidMount: function() {
+    GlossaryStore.addChangeListener(this._onChange);
+  },
+
+  componentWillUnmount: function() {
+    GlossaryStore.removeChangeListener(this._onChange);
+  },
+
+  _onChange: function() {
+    if (this.isMounted()) {
+      this.setState(this._getState());
+    }
+  },
+
+  _handleUpdate: function() {
+    Actions.updateGlossary(this.props.resId);
+  },
+
+  _handleCancel: function() {
+    Actions.resetEntry(this.props.resId);
+  },
+
+  render: function() {
+    var self = this;
+
+    if(this.props.resId === null || this.state.entry === null) {
+      return (<LoadingCell/>);
+    } else {
+      var info = (
+        <OverlayTrigger placement='top'
+          rootClose
+          overlay={<Tooltip id='src-info'>{self.props.info}</Tooltip>}>
+          <Icon className="cpri" name="info"/>
+        </OverlayTrigger>);
+
+      if(self.state.entry.status.isSaving === true) {
+        return (<div>{info} <Button kind='primary' className="ml1/4" loading>Update</Button></div>);
+      } else {
+        var deleteButton = null;
+
+        if(self.props.canDeleteEntry) {
+          deleteButton = (<DeleteEntry className='ml1/4' resId={self.props.resId} entry={self.state.entry}/>)
```

Comment:
parens not needed around this JSX, because it starts on the same line as the `=`